### PR TITLE
Remote Eval support

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
### Features and Changes

Enable remote evaluation
- Allow remote eval per SDK connection for supported SDKs (JS, React)
- Private features endpoint (required for a self-hosted remote eval service) at `/api/v1/sdk-payload/:key`
- Remote eval debugging endpoint at `/api/eval/:key`

### Screenshots

image

image

### CODE REVIEW

File: packages/back-end/src/controllers/features.ts [hunk 1]
```diff
@@ -153,11 +155,23 @@ export async function getFeaturesPublic(req: Request, res: Response) {
...
+      ssEvalEnabled,
       includeVisualExperiments,
       includeDraftExperiments,
       includeExperimentNames,
     } = await getPayloadParamsFromApiKey(key, req);
 
+    if (ssEvalEnabled) {
+      if (
+        !SECRET_API_KEY ||
+        req.headers["authorization"] !== `Bearer ${SECRET_API_KEY}`
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/controllers/features.ts:167]: use rest api with any api key

File: packages/back-end/src/app.ts [hunk 2]
```diff
@@ -203,6 +206,7 @@ app.get(
   featuresController.getFeaturesPublic
 );
 // For preflight requests
+// todo: do we actually need this?
 app.options(
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/app.ts:210]: this route seems unnecessary, as the above one already uses cors middelware

File: packages/back-end/src/routers/sdk-connection/sdk-connection.controller.ts [hunk 3]
```diff
@@ -56,10 +56,21 @@ export const postSDKConnection = async (
...
   }
 
+  let ssEvalEnabled = false;
+  if (orgHasPremiumFeature(org, "server-side-evaluation")) {
+    ssEvalEnabled = params.ssEvalEnabled || false;
+  }
+
+  if (ssEvalEnabled) {
+    encryptPayload = false;
+    // hashSecureAttributes = false;
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/routers/sdk-connection/sdk-connection.controller.ts:66]: todo: uncomment once the hashing PR goes live

File: packages/front-end/components/SyntaxHighlighting/Snippets/GrowthBookSetupCodeSnippet.tsx [hunk 4]
```diff
@@ -148,8 +149,7 @@ app.use(function(req, res, next) {
             encryptionKey
               ? `\n    decryptionKey: ${JSON.stringify(encryptionKey)}`
```
Comment (jdorn) [inline packages/front-end/components/SyntaxHighlighting/Snippets/GrowthBookSetupCodeSnippet.tsx:150]: This is missing a comma at the end and the newline missing below messes up the formatting

**Review by jdorn — APPROVED** [review]

**Pull request merged**